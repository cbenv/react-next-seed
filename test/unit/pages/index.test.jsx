import React from 'react'
import { render } from 'enzyme'
import { create } from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import HomePage from '../../../pages/index'

describe('Home Page', () => {
  const initialState = { counter: 0 }
  const mockStore = configureStore()
  let store
  let app
  let tree

  beforeEach(() => {
    store = mockStore(initialState)
    app = render(<HomePage store={store} />)
    tree = create(<HomePage store={store} />).toJSON()
  })

  it('consists of one header', () => {
    const header = app.find('.header')
    expect(header).toHaveLength(1)
  })

  it('consists of one counter', () => {
    const counter = app.find('.counter')
    expect(counter).toHaveLength(1)
  })

  it('renders to match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })
})
