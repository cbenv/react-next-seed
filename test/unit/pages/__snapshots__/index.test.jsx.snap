// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page renders to match snapshot 1`] = `
<div>
  <div
    className="header Header-root-1"
  >
    <header
      className="MuiPaper-root-11 MuiPaper-elevation4-17 MuiAppBar-root-3 MuiAppBar-positionStatic-7 MuiAppBar-colorPrimary-9"
    >
      <div
        className="MuiToolbar-root-38 MuiToolbar-regular-40 MuiToolbar-gutters-39"
      >
        <h2
          className="MuiTypography-root-42 MuiTypography-title-48 MuiTypography-colorInherit-61 Header-flex-2"
        >
          React Next Seed
        </h2>
      </div>
    </header>
  </div>
  <div
    className="counter"
  >
    <h1>
      0
    </h1>
    <button
      onClick={[Function]}
      type="button"
    >
      +1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      -1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
  </div>
</div>
`;
